sklearn/tests/test_common.py

```diff
@@ -144,6 +144,13 @@ def test_configure():
         os.chdir(setup_path)
         old_argv = sys.argv
         sys.argv = ['setup.py', 'config']
+
+        # OpenMP checks are build checks done before cythonization to allow
+        # fast fail. Since sklearn can be built and tested on 2 different
+        # machines, we disable it for a configure check.
```

Comment:
Since this addresses of very specific use-case, I think we need to be very clear on the explanation here. This has been confusing me for a while (I might still be missing something but feel free to correct me). Here's my attempt:

> This test will run every setup.py and eventually call check_openmp_support(), which tries to compile a C file that uses openmp, unless SKLEARN_NO_OPENMP is set. Some users might want to run the tests without having build-support for openmp. In particular, mac users need to set some env variables to build with openmp, which might not be set anymore at test time. We thus temporarily set SKLEARN_NO_OPENMP, so that the tests run smoothly.